distributed/deploy/adaptive_core.py

```diff
@@ -169,13 +187,13 @@ async def safe_target(self) -> int:
 
         return n
 
-    async def scale_down(self, n: int) -> None:
+    async def scale_down(self, workers: Iterable) -> None:
         raise NotImplementedError()
 
-    async def scale_up(self, workers: Iterable) -> None:
+    async def scale_up(self, n: int) -> None:
         raise NotImplementedError()
 
-    async def recommendations(self, target: int) -> dict:
+    async def recommendations(self, target: int) -> Recommendation:
```

Comment:
This is actually a breaking change since it is possible to subclass `Adaptive` and all consumers of `recommendations` would break.

Reply:
That's a very good point. I rewrote it into a `TypedDict`, so that it should remain backwards compatible while still improving readability.